nncf/common/quantization/quantizer_removal.py

```diff
@@ -36,13 +38,29 @@ def find_quantizer_nodes_to_cut(
     :param quantizer_node: The quantizer node that we want to remove.
     :param quantizer_metatypes: List of quantizer metatypes.
     :param const_metatypes: List of constant metatypes.
-    :param quantizable_metatypes: List of quantizable metatypes.
+    :param quantizable_metatypes: List of metatypes for operations
+        that may be quantized.
     :param quantize_agnostic_metatypes: List of quantize agnostic metatypes.
+    :param shape_of_metatypes: List of shape of metatypes.
     :return: A tuple (quantizer_nodes, ops) where
         - `quantizer_nodes` is the list of quantizer nodes
         - `ops` is the list of nodes that will be reverted to original precision
         if `quantizer_nodes` are removed.
     """
+    start_nodes = [*graph.get_input_nodes(), *graph.get_nodes_by_metatypes(quantizer_metatypes)]
+    queue = deque(start_nodes)
+    visited = {x.node_name: True for x in start_nodes}
+
+    while len(queue) != 0:
```

Comment:
It looks like, this is the logic for removing the shapeof subgraph. If I'm right, could you collaborate with @KodiaqQ to create a standalone generic method for removing shapeof subgraphs.

Reply:
Done